torchao/experimental/ops/mps/register.mm

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+// All rights reserved.
+//
+// This source code is licensed under the license found in the
+// LICENSE file in the root directory of this source tree.
+
+// clang-format off
+#include <torch/extension.h>
+#include <ATen/native/mps/OperationUtils.h>
+#include <torchao/experimental/kernels/mps/src/lowbit.h>
+// clang-format on
+
+namespace torchao {
+namespace kernels {
+namespace mps {
+namespace lowbit {
+namespace aten {
+
+using Tensor = at::Tensor;
+using namespace at::native::mps;
+
+// LowBit Quantized Linear on MPS Backend
+template <int nbit>
+Tensor linear_mps_kernel(
+    const Tensor& A,
+    const Tensor& B,
+    int64_t group_size,
+    const Tensor& SZ) {
+  auto M = A.size(0);
+  auto N = B.size(0);
+  auto K = A.size(1);
+  auto C = at::empty({M, N}, A.options());
+
+  LowBitQuantWeights<nbit>::linear(
+      getMTLBufferStorage(A),
+      getMTLBufferStorage(B),
+      group_size,
+      getMTLBufferStorage(SZ),
+      getMTLBufferStorage(C),
+      M,
+      K,
+      N,
+      scalarToMetalTypeString(A));
+
+  return C;
+}
+
+// LowBit Packing on CPU Backend
+template <int nbit>
+Tensor pack_weights_cpu_kernel(const Tensor& W) {
+  auto N = W.size(0);
+  auto K = W.size(1);
+  auto B = at::empty({N, nbit * K / 8}, W.options());
+
+  uint8_t* w_ptr = W.data_ptr<uint8_t>();
+  uint8_t* b_ptr = B.data_ptr<uint8_t>();
+
+  LowBitQuantWeights<nbit>::pack(w_ptr, b_ptr, N, K);
+
+  return B;
+}
+
+// Registers _C as a Python extension module.
+PYBIND11_MODULE(TORCH_EXTENSION_NAME, m) {}
```

Comment:
Why do you need this?

Reply:
I believe this piece is what allows me to do: `import torchao_mps_ops`, instead of having to load the library path